homeassistant/components/bosch_alarm/config_flow.py

```diff
@@ -152,11 +189,15 @@ async def async_step_auth(
                 _LOGGER.exception("Unexpected exception")
                 errors["base"] = "unknown"
             else:
-                if serial_number:
+                if serial_number and not self.unique_id:
                     await self.async_set_unique_id(str(serial_number))
-                    self._abort_if_unique_id_configured()
-                else:
-                    self._async_abort_entries_match({CONF_HOST: self._data[CONF_HOST]})
+                if self.source == SOURCE_USER:
+                    if serial_number:
+                        self._abort_if_unique_id_configured()
+                    else:
+                        self._async_abort_entries_match(
+                            {CONF_HOST: self._data[CONF_HOST]}
+                        )
```

Comment:
I think we should be careful here with the unique id, as we both have a serial number as unique id and a mac address now.

So a few points I want to mention:
1. Getting the mac address is nice, as it allows us to add it to the device info, allowing the device to merge with a device created by a network integration (like unifi or netgear).
2. Since not all panels have a serial number, having the mac address is nice since we could add the mac address after setting up, and we can migrate the entities to use that mac address.

So I think I would recommend to keep the serial number as the unique id, but store the mac address in the entry data.

How does the device connect to the ethernet?

Reply:
some of the panels have ethernet built in, some of them have a seperate ip module